pyro/distributions/iaf.py

```diff
@@ -18,7 +17,7 @@ class InverseAutoregressiveFlow(Transform):
     Example usage:
 
     >>> base_dist = dist.Normal(torch.zeros(10), torch.ones(10))
-    >>> iaf = InverseAutoregressiveFlow(10, 40)
+    >>> iaf = InverseAutoregressiveFlow(AutoRegressiveNN(10, [40]))
```

Comment:
iaf needs updated docstrings. in particular it should be clear it requires an autoregressive_nn of a particular sort (e.g. count_params=2)

Reply:
Done!